Hunting Queries/DnsEvents/DNS_HighNXDomainCount.txt

```diff
@@ -1,24 +1,82 @@
+let thirdQT = quartileFunction(75, 7d, 2d) | project-rename percentile_SearchList_75 = qPercentiles;
+let threshold = (firstQT
+| join thirdQT on ClientIP
+| extend IPthreshold = percentile_SearchList_75 + (1.5*exp(3)*(percentile_SearchList_75 - percentile_SearchList_25))
+| project ClientIP, IPthreshold);
+let Tab1 = (
+nxDomainDnsEvents
+| where TimeGeneratedUtc > ago(1d)
+| summarize TotalNXLookups=dcount(truncatedDomain) by ClientIP
+| sort by TotalNXLookups desc
+| join ['threshold'] on ClientIP
+| where TotalNXLookups > IPthreshold // Comment this line in order to view results filtered by Global Threshold only. 
+| join kind = leftouter (nxDomainDnsEvents
+    | where TimeGeneratedUtc > ago(1d)
+    | extend SearchList = iff((strlen(Name) - indexof(Name, tostring(split(Name, ".")[-2])) ) >= 7, strcat(tostring(split(Name, ".")[-2]), ".", tostring(split(Name, ".")[-1])) , strcat(tostring(split(Name, ".")[-3]), ".", tostring(split(Name, ".")[-2]), ".", tostring(split(Name, ".")[-1])))
+    | summarize domainCount=count() by SearchList, ClientIP
+    | sort by domainCount desc
+    | project SearchList = strcat(SearchList," (",tostring(domainCount),")"), ClientIP
+    ) on ClientIP
+| summarize SLDs_TotalLookups = makelist(SearchList) by ClientIP, TotalNXLookups, IPthreshold
+| sort by TotalNXLookups desc);
+//
+let quartileFunction2 = view (mypercentile:long, startTimeSpan:timespan) {
+(nxDomainDnsEvents
+| where TimeGeneratedUtc > ago(startTimeSpan)
+| summarize domainCount = dcount(truncatedDomain) by ClientIP, bin(TimeGeneratedUtc, startTimeSpan)
+| summarize event_count = count() by domainCount
+| sort by domainCount asc
+| summarize perc2 = percentilesw(domainCount, event_count, mypercentile));
+};
+let firstQ = toscalar(quartileFunction2(25, 1d));
+let thirdQ = toscalar(quartileFunction2(75, 1d));
+let GlobalThreshold = toscalar(thirdQ + (1.5*exp(3)*(thirdQ - firstQ)));
+let Tab2 = (
 nxDomainDnsEvents
-| extend sld = tostring(split(Name, ".")[-2])
-| summarize dcount(sld) by ClientIP
-| where dcount_sld > 100
-| join kind = inner (nxDomainDnsEvents | summarize by Name, ClientIP) on ClientIP
-| summarize sampleNXDomainList=makelist(Name, 10)  by ClientIP, dcount_sld
+| where TimeGeneratedUtc > ago(1d)
+| summarize TotalNXLookups = dcount(truncatedDomain) by ClientIP
+| sort by TotalNXLookups desc
+| where TotalNXLookups > GlobalThreshold // Comment this line in order to view results filtered by IPThreshold only. 
+| join kind = leftouter (nxDomainDnsEvents
+    | where TimeGeneratedUtc > ago(1d)
+    | summarize domainCount=dcount(Name) by truncatedDomain, ClientIP
+    | sort by domainCount desc
+    | project truncatedDomain= strcat(truncatedDomain," (",tostring(domainCount),")"), ClientIP
+    ) on ClientIP
+| summarize SLDs_TotalLookups = makelist(truncatedDomain), UniqueSLDsCount=count(truncatedDomain) by ClientIP, TotalNXLookups, GlobalThreshold
+| sort by TotalNXLookups desc);
+Tab1
```

Comment:
maybe we could give more self-explanatory names to quartfunction1, quartileFunction2, Tab1 and Tab2